cli/ops.rs

```diff
@@ -2194,3 +2201,135 @@ fn op_get_random_values(
   thread_rng().fill(&mut data.unwrap()[..]);
   Box::new(ok_future(empty_buf()))
 }
+
+fn op_dynamic_lib_load(
+  state: &ThreadSafeState,
+  base: &msg::Base<'_>,
+  _data: Option<PinnedBuf>,
+) -> Box<OpWithError> {
+  let cmd_id = base.cmd_id();
+  let inner = base.inner_as_dynamic_lib_load().unwrap();
+  let (filename, filename_) = match resolve_path(inner.filename().unwrap()) {
+    Err(err) => return odd_future(err),
+    Ok(v) => v,
+  };
+
+  if let Err(e) = state.check_dlopen(&filename_) {
+    return odd_future(e);
+  }
+
+  Box::new(futures::future::result(move || -> OpResult {
+    let lib = bindings::load_binding(filename)?;
+    let lib_id: bindings::DylibId =
+      state.next_dylib_id.fetch_add(1, Ordering::SeqCst);
```

Comment:
I think this should be a Resource...

Reply:
My main concern with that is the global nature of resources.

Reply:
Resources are currently global, but I intend to make them per-worker in the future. (or even per-module...). Basically I'd like each Context to have a security token, which gives them access to Permissions struct and Resources table.

Reply:
Another reason for not to do this is performance. The resource table is currently using a `Mutex` instead of a `RwLock`. The bulk of the runtime operations for this system after load will be read only, and thus benefit greatly from a `RwLock`.

Reply:
We haven't been able to measure any performance problem with the resource table.

The dylib is something that is allocated outside of JS, which is managed by JS. If we hope to be able to clean up these things we need to handle them in a uniform way. Ideally when worker terminates it drops its resources too (this probably doesn't happen currently, but that's the idea).